src/file_parsing.jl

```diff
@@ -151,35 +151,68 @@ function log_component!(
     end
 end
 
+"""
+    _get_typed_value(pd,val,valtype)
+
+enforces `val` to be of type as specified in the toml file
+- `float_type(pd)` if type=\"float\"
+- `Int` if type=\"integer\"
+- `String` if  type=\"string\"
+Default type of `String` is used if no type is provided.
+"""
+function _get_typed_value(pd, val, valtype)
+
+    if valtype == "float"
+        return float_type(pd)(val)
+    elseif valtype == "integer"
+        return Int(val)
+    elseif valtype == "string"
+        return String(val)
+    else
+        error(
+            "Value type ",
+            valtype,
+            "not recognised, please select from: \"string\", \"float\", or \"integer\" ",
+        )
+    end
+end
+
+
+
 """
     get_values(pd::AbstractTOMLDict, names)
 
 gets the `value` of the named parameters.
 """
 function get_values(pd::AliasParamDict, aliases::NAMESTYPE)
-    FT = float_type(pd)
     data = pd.data
     # TODO: use map
     ret_values = []
     for alias in aliases
         for (key, val) in data
             alias ≠ val["alias"] && continue
             param_value = val["value"]
+            param_type = haskey(val, "type") ? val["type"] : "string"
             elem =
-                eltype(param_value) != FT ? map(FT, param_value) : param_value
+                isa(param_value, AbstractVector) ?
+                map(x -> _get_typed_value(pd, x, param_type), param_value) :
+                _get_typed_value(pd, param_value, param_type)
```

Comment:
```suggestion
                map(x -> _get_typed_value(pd, x, param_type), param_value)
```
I think we might be able to just do this? can you try it out?

Reply:
I don't use `eltype` in `_get_typed_value` so not sure - I'll give it a go

Reply:
Ahh it actually has an issue with `string` - it tries to map it to each individual character

Reply:
Ah, right, the original is fine.